perfkitbenchmarker/log_util.py

```diff
@@ -0,0 +1,144 @@
+# Copyright 2014 Google Inc. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Utilities related to loggers and logging.
+"""
+
+from contextlib import contextmanager
+import logging
+import threading
+
+DEBUG = 'debug'
+INFO = 'info'
+LOG_LEVELS = {
+    DEBUG: logging.DEBUG,
+    INFO: logging.INFO
+}
+
+
+thread_local = threading.local()
+
+
+class ThreadLogContext(object):
+  """Per-thread context for log message prefix labels."""
+  def __init__(self, thread_log_context=None):
+    """Constructs a ThreadLogContext by copying a previous ThreadLogContext.
+
+    Args:
+      thread_log_context: A ThreadLogContext for an existing thread whose state
+        will be copied to initialize a ThreadLogContext for a new thread.
+    """
+    if thread_log_context:
+      self._label_list = thread_log_context._label_list[:]
+    else:
+      self._label_list = []
+    self._RecalculateLabel()
+
+  @property
+  def label(self):
+    return self._label
+
+  def _RecalculateLabel(self):
+    """Recalculate the string label used to to prepend log messages.
+
+    The label is the concatenation of all non-empty strings in the _label_list.
+    """
+    non_empty_string_list = [s for s in self._label_list if s]
+    if len(non_empty_string_list):
+      self._label = ' '.join(non_empty_string_list) + ' '
+    else:
+      self._label = ''
+
+  @contextmanager
+  def LabelExtension(self, label_extension):
```

Comment:
Please document this method.  

Consider a more informative name (preferably stating an action).